site/docs/how-to-release.md

```diff
@@ -323,11 +323,11 @@ Please follow the instructions on the GitHub repository in the [`README.md` in t
 
 #### Versioned Docs
 
-The versioned docs in the source tarball are located under `site/docs`. Check out the `docs` branch in the `iceberg` repo and copy
```

Comment:
While doing version docs release, I observed that there is no version doc present in the release tar. We should copy from release tag and from docs instead of site/docs